huaweicloud/services/cci/resource_huaweicloud_cciv2_namespace.go

```diff
@@ -0,0 +1,288 @@
+
+// @API CCI DELETE /apis/cci/v2/namespaces/{name}
+// @API CCI GET /apis/cci/v2/namespaces/{name}
+// @API CCI POST /apis/cci/v2/namespaces
+func ResourceNamespace() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNamespaceCreate,
+		UpdateContext: resourceNamespaceUpdate,
+		ReadContext:   resourceNamespaceRead,
+		DeleteContext: resourceNamespaceDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		CustomizeDiff: config.FlexibleForceNew(namespaceNonUpdatableParams),
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The name of the namespace.`,
+			},
+			"annotations": {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: `The annotations of the namespace.`,
+			},
+			"labels": {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: `The labels of the namespace.`,
+			},
+			"creation_timestamp": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The creation timestamp of the namespace.`,
+			},
+			"resource_version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The resource version of the namespace.`,
+			},
+			"uid": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The uid of the namespace.`,
+			},
+			"api_version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The API version of the namespace.`,
+			},
+			"kind": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The kind of the namespace.`,
+			},
+			"finalizers": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+				Description: `The finalizers of the namespace.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The status of the namespace.`,
+			},
+			"enable_force_new": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"true", "false"}, false),
+				Description:  utils.SchemaDesc("", utils.SchemaDescInput{Internal: true}),
+			},
+		},
+	}
+}
+
+func resourceNamespaceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.CciV2Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("Error creating CCI v2 client: %s", err)
+	}
```

Comment:
cfg := meta.(*config.Config)
	region := cfg.GetRegion(d)
	attachClient, err := cfg.NewServiceClient("cci", region)
	if err != nil {
		return diag.Errorf("error creating CCI client: %s", err)
	}

and fix other places